spec/options_spec.rb

```diff
@@ -1,40 +1,148 @@
 # typed: false
 describe RedisMemo::Options do
   context 'cache validation' do
-    it 'validates the cache result' do
-      allow(RedisMemo::DefaultOptions).to receive(:cache_validation_sampler) do
-        proc { true }
-      end
+    def allow_rand_return(value)
+      allow(Random).to receive(:rand).and_return(value)
+    end
 
+    def expect_count(orig_value, updated_value)
```

Comment:
Is `count` just a mechanism to tell whether or not the :cache_out_of_date_handler is called? This is confusing because the klass we're testing on also has a count instance variable. Could we just do something like `expect(RedisMemo::DefaultOptions).to recieve(:cache_out_of_date_handler)`?

This isn't just specific to your test though, I think this is used elsewhere in this class.

Reply:
> Could we just do something like expect(RedisMemo::DefaultOptions).to recieve(:cache_out_of_date_handler)?

Hmm I guess one of the reasons we had 
```
allow(RedisMemo::DefaultOptions).to receive(:cache_out_of_date_handler) do
        proc { count += 1 }
 end
```

so i think the `porc {count += 1} still adds value 
was because we not only want to make sure it receives the `cache_out_of_date_handler`, but also exectutes the blocks in 
```
config.cache_out_of_date_handler = proc do ...
    ...
  end
```

> This is confusing because the klass we're testing on also has a count instance variable.

if this is just a confusion of the names, then i think we could simply use another variable lol

let me know how you think about it!